js/main.js

```diff
@@ -1,3 +1,22 @@
+function goToNetworkAlert() {
+  switchView('network-error');
+}
+
+function testConnectivity() {
+  var newReq = new XMLHttpRequest();
+  newReq.open('GET', 'https://random-words-api.vercel.app/word');
+  newReq.onreadystatechange = function () {
+    if (newReq.readyState === XMLHttpRequest.DONE) {
+      var status = newReq.status;
+      if (status !== 200) {
+        goToNetworkAlert();
+      }
```

Comment:
The response status should be checked during each of your XHRs. Just checking the connection one time when the page loads doesn't make much sense, because the user probably has a fine connection at first if the page even loaded to begin with.

Also, iirc the XHR object can dispatch `'error'` events in the case of disconnection mid-request.